### PR TITLE
feat(impression-area): Merge internal changes

### PR DESCRIPTION
**Pull request by ragingwind**
## Overview

Merge latest code and test

## PR Checklist

- [ ] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/impression-area/src/useImpressionRef.ko.md [hunk 1]
```diff
@@ -17,10 +18,6 @@ const ref = useImpressionRef({
   // 요소가 브라우저 뷰포트에서 나왔을 때 호출되는 callback
   onImpressionEnd: () => {},
 
-  // 타겟 요소의 가시성을 검사하기 위해 Viewport 대신 사용할 요소를 지정할 수 있습니다.
-  // @default null
-  root={root}
-
```
Comment (raon0211) [inline packages/react/impression-area/src/useImpressionRef.ko.md:23]: Is this change intended here?

File: packages/react/impression-area/src/useImpressionRef.tsx [hunk 2]
```diff
@@ -2,73 +2,98 @@
 import { usePreservedCallback, useRefEffect } from '@toss/react';
 import { noop } from '@toss/utils';
 import debounce from 'lodash.debounce';
-import { useMemo, useRef } from 'react';
+import { useCallback, useEffect, useMemo, useRef } from 'react';
 import { ImpressionAreaProps } from './ImpressionArea';
 
 type Options = Pick<
   ImpressionAreaProps,
-  'rootMargin' | 'onImpressionStart' | 'onImpressionEnd' | 'timeThreshold' | 'areaThreshold' | 'root'
```
Comment (raon0211) [inline packages/react/impression-area/src/useImpressionRef.tsx:10]: Is this change intended?

Comment (ssi02014) [reply packages/react/impression-area/src/useImpressionRef.tsx:10]: https://github.com/toss/slash/pull/283 
Previously, I created a pull request to add the "root" option.
I would appreciate it if you could check that pull request. 
(Cause of the problem: `iframe`)
@ragingwind

**Review by raon0211 — APPROVED** [review]
👍
